### PR TITLE
Add fast path for forwarddiff_color_jacobian! with sparse J

### PR DESCRIPTION
**Pull request by sjdaines**
Partial fix for https://github.com/JuliaDiff/SparseDiffTools.jl/issues/138 https://github.com/JuliaDiff/SparseDiffTools.jl/issues/100

Add a fast path for the case where J and sparsity are are both SparseMatrixCSC
and have the same sparsity pattern.

### CODE REVIEW

File: src/differentiation/compute_jacobian_ad.jl [hunk 1]
```diff
@@ -316,3 +333,17 @@ function forwarddiff_color_jacobian!(J::AbstractMatrix{<:Number},
     end
     return J
 end
+
+# fast version of FiniteDiff._colorediteration! for the case where J and sparsity have the same sparsity pattern
+@inline function _colorediteration!(Jsparsity::SparseMatrixCSC,vfx,colorvec,color_i,ncols)
```
Comment (ChrisRackauckas) [inline src/differentiation/compute_jacobian_ad.jl:338]: Shouldn't this just go to FiniteDiff.jl?

**Comment by sjdaines** [comment]
Probably... (I admit I don't I understand how this should work and feel like I'm proliferating cut-and-paste cases instead of sane dispatches).

PR https://github.com/JuliaDiff/FiniteDiff.jl/pull/124

Now we have:

1. add a _use_sparseCSC_common_sparsity!(J, sparsity) test function to FiniteDiff.iteration_utils.
2. move the new _colorediteration!(Jsparsity::SparseMatrixCSC,vfx,colorvec,color_i,ncols) to FiniteDiff.iteration_utils
3. FiniteDiff.finite_difference_jacobian! uses the same optimisation (although doesn't look like anyone has hit this problem there?)

File: src/differentiation/compute_jacobian_ad.jl (hunk 1 above)
Comment (sjdaines) [reply src/differentiation/compute_jacobian_ad.jl]: PR JuliaDiff/FiniteDiff.jl#124

**Comment by ChrisRackauckas** [comment]
Add a lower bound on 2.8.1

**Pull request merged**